src/main/java/org/sonarsource/sonarlint/ls/util/Utils.java

```diff
@@ -273,4 +275,18 @@ public static boolean isDelegatingIssueWithServerIssueKey(String serverIssueKey,
     return issueEntry.getValue().getIssue() instanceof DelegatingIssue
       && (serverIssueKey.equals(((DelegatingIssue) issueEntry.getValue().getIssue()).getServerIssueKey()));
   }
+
+  public static void runIfAnalysisNeeded(String uri, SonarLintExtendedLanguageClient client,
+    LanguageClientLogger lsLogOutput, Runnable analyse) {
+    client.shouldAnalyseFile(new SonarLintExtendedLanguageServer.UriParams(uri)).thenAccept(checkResult -> {
+      if (Boolean.TRUE.equals(checkResult.isShouldBeAnalysed())) {
+        analyse.run();
+      } else {
+        var reason = checkResult.getReason();
+        if (reason != null) {
+          lsLogOutput.info(reason + " " + uri);
```

Comment:
Minor comment - put `uri` into double-quotes and it will make file uri clickable in the logs.